drivers/video/video_esp32_dvp.c

```diff
@@ -344,6 +395,19 @@ static void video_esp32_cam_ctrl_init(const struct device *dev)
 	cam_ll_enable_invert_hsync(data->hal.hw, cfg->invert_hsync);
 }
 
+#ifdef CONFIG_VIDEO_ESP32_VSYNC_INTERRUPT
+static int video_esp32_enable_interrupt(const struct device *dev)
+{
+	const struct video_esp32_config *cfg = dev->config;
+	struct video_esp32_data *data = dev->data;
+
+	cam_ll_clear_interrupt_status(data->hal.hw, VIDEO_ESP32_VSYNC_MASK);
+	cam_ll_enable_interrupt(data->hal.hw, VIDEO_ESP32_VSYNC_MASK, true);
+	return esp_intr_alloc(cfg->irq_source, cfg->irq_priority,
+			      (intr_handler_t)video_esp32_irq_handler, (void *)dev, NULL);
```

Comment:
I appreciate the compactness, but `esp_intr_alloc` returns different error codes from Zephyr.
https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/system/esp_err.html?highlight=esp_err_t#c.ESP_OK

This means that a function checking `if (err == -EAGAIN) { try_again(); }` could get confused by the error code from ESP32 mismatching those from `errno.h`. Would it make sense to switch it to a selected error? Such as `-ENOMEM` since it is about allocation.

Reply:
> I appreciate the compactness, but `esp_intr_alloc` returns different error codes from Zephyr.

I don't think so. `esp_intr_alloc` is a zephyr function returning `-EINVAL` or `-ENODEV` in case of an error. 
https://github.com/zephyrproject-rtos/zephyr/blob/1fe904539d32b5b7d18361d63acc7a47c9171400/include/zephyr/drivers/interrupt_controller/intc_esp32.h#L165-L169
https://github.com/zephyrproject-rtos/zephyr/blob/1fe904539d32b5b7d18361d63acc7a47c9171400/drivers/interrupt_controller/intc_esp32.c#L704

Reply:
I must have looked at the wrong place. Thank you for checking and sorry for the noise!

Reply:
Oh, an ESP-IDF function of the same name! :laughing:
https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/system/intr_alloc.html#_CPPv414esp_intr_allocii14intr_handler_tPvP13intr_handle_t